test/controllers/api/v1/ping_thing_controller_test.rb

```diff
@@ -125,11 +125,44 @@ def setup
     assert_equal 2, response_to_json(@response.body).size
   end
 
-  test "GET api_v1_ping_things with limit returns correct pings starting from the newest" do
-    create(:ping_thing, :testnet, response_time: 123)
+  test "GET api_v1_ping_things with limit returns correct pings starting from the newest reported_at" do
+    create(:ping_thing, :testnet, reported_at: 2.day.ago, response_time: 123)
+    create(:ping_thing, :testnet, reported_at: 1.day.ago, response_time: 321)
 
     get api_v1_ping_things_path(network: "testnet", limit: 1), headers: { "Token" => @user.api_token }
 
+    json = response_to_json(@response.body)
+    assert_equal 321, json.first["response_time"]
+  end
+
+  test "GET api_v1_ping_things without page present returns the first page" do
+    3.times do
+      create(:ping_thing, :testnet)
+    end
+
+    get api_v1_ping_things_path(network: "testnet", limit: 2), headers: { "Token" => @user.api_token }
+
+    assert_response 200
+    assert_equal 2, response_to_json(@response.body).size
+  end
+
+  test "GET api_v1_ping_things with page param, returns correct pings starting from the newest reported_at" do
+    create(:ping_thing, :testnet, reported_at: 3.day.ago, response_time: 123)
+    create(:ping_thing, :testnet, reported_at: 2.day.ago, response_time: 456)
+    create(:ping_thing, :testnet, reported_at: 1.day.ago, response_time: 789)
+
+    get api_v1_ping_things_path(network: "testnet", limit: 1, page: 1), headers: { "Token" => @user.api_token }
+
+    json = response_to_json(@response.body)
+    assert_equal 789, json.first["response_time"]
```

Comment:
You can also check size of the array to be sure that it returned exactly one element.